workshop/content/20-typescript/30-hello-cdk/300-cdk-watch.md

```diff
@@ -0,0 +1,179 @@
+
+First, as an exercise, change your lambda code in `lambda/hello.js` slightly and 
+time how long it takes to run `cdk deploy`. It will help us get a baseline for how 
+long `cdk deploy` takes:
+
+```bash
+time cdk deploy
+```
+
+The output will look something like this:
+
+```
+INSERT OUTPUT HERE
+```
+
+## CDK Watch
+
+{{% notice info %}} This command deliberately introduces drift in CloudFormation 
+stacks in order to speed up deployments. For this reason, only use it for 
+development purposes. Never use CDK Watch for your production deployments!
+{{% /notice %}}
+
+We can speed up that deployment time with `cdk watch`, which watches your
+CDK application and looks for hotswappable changes. Which `cdk watch` finds a
+change that can be hotswapped, it bypasses the CloudFormation deployment entirely
+and instead makes the necessary SDK call to update your code for you. Changes to 
+your AWS Lambda asset code, AWS Step Function State Machine definition, and other
+non-resource changes have the potential to be hotswapped. Here, we will try
+`cdk watch` with your AWS Lambda asset code.
+
+## Modify your `cdk.json` file
+
+When the `cdk watch` command runs, the files that it observes is determined by the
+`"watch"` setting in the `cdk.json` file. It has two sub-keys, `"include"` and
+`"exclude"`, each of which can be either a single string or an array of strings.
+Each entry is interpreted as a path relative to the location of the `cdk.json`
+file. Globs, both `*` and `**`, are allowed to be used.
+
+Your `cdk.json` file should look like this:
+
+```json
+{
+  "app": "npx ts-node --prefer-ts-exts bin/workshop.ts",
+  "watch": {
+    "include": [
+      "**"
+    ],
+    "exclude": [
+      "README.md",
+      "cdk*.json",
+      "**/*.d.ts",
+      "**/*.js",
+      "tsconfig.json",
+      "package*.json",
+      "yarn.lock",
+      "node_modules",
+      "test"
+    ]
+  },
+  "context": {
+    ...
+  }
+}
+```
+
+As you can see, the sample app comes with a suggested `"watch"` setting. We do in
+fact want to observe our `.js` files in the `lambda` folder, so let's remove
+`"**/*.js"` from the `"exclude"` list:
+
+```json
+{
+  "app": "npx ts-node --prefer-ts-exts bin/workshop.ts",
+  "watch": {
+    "include": [
+      "**"
+    ],
+    "exclude": [
+      "README.md",
+      "cdk*.json",
+      "**/*.d.ts",
+      "tsconfig.json",
+      "package*.json",
+      "yarn.lock",
+      "node_modules",
+      "test"
+    ]
+  },
+  "context": {
+    ...
+  }
+}
+```
+
+Now you're all set to start watching!
+
+## See how long 'CDK Watch' takes
+
+First, call `cdk watch`: 
+
+```bash
+cdk watch
+```
+
+This will trigger an initial deployment and immediately begin observing the files
+you've specified in `cdk.json`.
+
+For the purposes of this exercise, make sure that the initial deployment is
+finished before proceeding. Otherwise, the changes you make will be queued for
+deployment and released _after_ the first deployment finishes.
```

Comment:
Not sure we need this paragraph - even if the changes come during, they will still be picked up, so I don't think it makes sense to add this disclaimer.